exporters/otlp/testing-internal/src/main/java/io/opentelemetry/exporter/otlp/testing/internal/AbstractHttpTelemetryExporterTest.java

```diff
@@ -655,6 +658,39 @@ void nonRetryableError(int code) {
     assertThat(attempts).hasValue(1);
   }
 
+  @Test
+  void proxy() {
+    // configure mockserver to proxy to the local OTLP server
+    InetSocketAddress serverSocketAddress = server.httpSocketAddress();
+    try (ClientAndServer clientAndServer =
+        ClientAndServer.startClientAndServer(
+            serverSocketAddress.getHostName(), serverSocketAddress.getPort())) {
+      TelemetryExporter<T> exporter =
+          exporterBuilder()
+              // Configure exporter with server endpoint, and proxy options to route through
+              // mockserver proxy
+              .setEndpoint(server.httpUri() + path)
+              .setProxyOptions(
+                  ProxyOptions.create(
+                      InetSocketAddress.createUnresolved("localhost", clientAndServer.getPort())))
+              .build();
+
+      try {
+        List<T> telemetry = Collections.singletonList(generateFakeTelemetry());
+
+        assertThat(exporter.export(telemetry).join(10, TimeUnit.SECONDS).isSuccess()).isTrue();
+        // assert that mock server received request
+        assertThat(clientAndServer.retrieveRecordedRequests(new org.mockserver.model.HttpRequest()))
+            .hasSize(1);
+        // assert that server received telemetry from proxy, and is as expected
+        List<U> expectedResourceTelemetry = toProto(telemetry);
+        assertThat(exportedResourceTelemetry).containsExactlyElementsOf(expectedResourceTelemetry);
```

Comment:
If the proxy was ignored completely in the implementation (eg. `setProxyOptions()` was a noop) would this test still pass? If so, is there a way to validate that the proxy was actually used, not just that the data made it across?

Reply:
> If the proxy was ignored completely in the implementation (eg. setProxyOptions() was a noop) would this test still pass? 

No the test fails. Line 683 retrieve requests recorded by the proxy and verifies it received the expected request. If the implementation fails to route through the configured proxy, the mockserver proxy will never receive the request and line 683 fails.